x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_overview_logic.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { CrawlerData, CrawlerDataFromServer, CrawlerDomain } from './types';
+import { crawlerDataServerToClient } from './utils';
+
+interface CrawlerOverviewValues {
+  domains: CrawlerDomain[];
+}
+
+interface CrawlerOverviewActions {
+  fetchCrawlerData(): void;
+  setCrawlerData(data: CrawlerData): { data: CrawlerData };
+}
+
+export const CrawlerOverviewLogic = kea<
+  MakeLogicType<CrawlerOverviewValues, CrawlerOverviewActions>
+>({
+  path: ['enterprise_search', 'app_search', 'crawler', 'crawler_overview'],
+  actions: {
+    fetchCrawlerData: () => null,
+    setCrawlerData: (data) => ({ data }),
+  },
+  reducers: {
+    domains: [
+      [],
+      {
+        setCrawlerData: (_, { data: { domains } }) => domains,
```

Comment:
I've stated this on other occasions and I'll mention it again at the risk of sounding like a broken record... I think it's preferable to favor event oriented actions rather than explicit "setter" type actions.

This would be `onCrawlerDataLoaded` to be consistent with other recent logic files.

Reply:
I've done this in d8eaf85